src/common/hooks/useQLDB.ts

```diff
@@ -0,0 +1,63 @@
+import { QldbDriver } from "amazon-qldb-driver-js/dist/src/QldbDriver";
+import { useCallback, useContext, useState } from "react";
+
+import QLDBSession = require("aws-sdk/clients/qldbsession");
+import { ClientConfiguration } from "aws-sdk/clients/qldbsession";
+import { AppStateContext } from "../../core/AppStateProvider";
+
+interface QLDBResult<T> {
+  results: T[];
+  error?: Error;
+  query: () => Promise<void>;
+  clear: () => void;
+}
+
+export function useQLDB<T>(
+  ledger: string,
+  lambda: (driver: QldbDriver) => Promise<T[]>
+): QLDBResult<T> {
+  const [results, setResults] = useState<T[]>([]);
+  const [error, setError] = useState<Error | undefined>();
+
+  const [
+    {
+      credentials: { accessKeyId, secretAccessKey },
+      region,
+    },
+  ] = useContext(AppStateContext);
+
+  const clear = useCallback(() => {
+    setResults([]);
+    setError(undefined);
+  }, []);
+
+  const executeQuery = useCallback(async () => {
+    if (!accessKeyId || !secretAccessKey) {
+      setError(new Error("Missing credentials"));
+      return;
+    }
+
+    const driver = new QldbDriver(
```

Comment:
If I understand this correctly, this creates a new driver for every query we execute. @allimn can probably provide better info but in the console we store drivers by ledger name in a Context so components can reuse drivers (see DriverContext.tsx in our console package). Ang can correct me if I'm wrong, but I think this was important because constructing a new driver is relatively expensive.

I think we need to do something similar here. Maybe the editor component should take a DriverProvider parameter or something along those lines.

Reply:
Yes, there is a session pool maintained in a driver. We usually want to reuse the session pool(driver) so that driver won't start session every time when we execute the query.

Reply:
Good point! I'll create a Context for the drivers. 
We can discuss if It makes sense to do this in the driver itself though, like a Singleton per ledger, so if you instanciate a new driver with the same region/ledger/params it reuses the previous one - similar to how the DriverContet.tsx does.